src/actions/append.ts

```diff
@@ -1,24 +1,28 @@
 import { EOL } from 'os';
 import type { MemFsEditor } from '../index.js';
 
-export type AppendOptions = { create?: boolean; trimEnd?: boolean; separator?: string };
-
-export default function append(this: MemFsEditor, to: string, contents: string | Buffer, options?: AppendOptions) {
+export default function append(
+  this: MemFsEditor,
+  to: string,
+  contents: string | Buffer,
+  options?: { create?: boolean; trimEnd?: boolean; separator?: string },
+) {
   const opts = {
+    create: false,
     trimEnd: true,
     separator: EOL,
     ...options,
   };
 
-  if (!this.exists(to) && opts.create) {
-    this.write(to, contents);
-    return;
+  if (!this.exists(to) && !opts.create) {
+    throw new Error(`${to} doesn't exist`);
   }
 
-  let currentContents = this.read(to);
-  if (currentContents && opts.trimEnd) {
-    currentContents = currentContents.replace(/\s+$/, '');
+  let currentContent = this.read(to, { defaults: '' });
```

Comment:
defaults should not be passed because it should not happen.
A file with null content can mean that it doesn't exist in disk or it has been removed in mem-fs.
Append should throw in both cases.

Reply:
The `create` option AFAIK is there to allow creating new files upon append?

So the exist check first ensure we should proceed with creating the file (if not, it throws so doesn't proceed). For the rest of the function, I think it's simpler to have a single call to `write` than the previous implementation following 2 code path.